core/lib/multivm/src/versions/vm_1_3_2/refunds.rs

```diff
@@ -36,11 +36,10 @@ impl<H: HistoryMode, S: WriteStorage> VmInstance<S, H> {
         // For now, bootloader charges only for base fee.
         let effective_gas_price = self.block_context.base_fee;
 
-        let bootloader_eth_price_per_pubdata_byte = U256::from(effective_gas_price)
```

Comment:
You should avoid changing multivm version except for latest. Instead you should glue results

Reply:
The problem here is we have updated the `L1PeggedBatchFeeModelInput` struct to move gas prices to `U256`. This struct is defined on the `zksync_types` crate under fee model, outside the vm/multivm stuff. This change breaks all the `multivm` methods/functions that use it because they expect  their fields to still be `u64` (for example the `derive_base_fee_and_gas_per_pubdata` function).

I'm not sure what the best way to proceed here is; we can introduce a new struct (something like `L1PeggedBatchFeeModelInputU256`) that the latest vm uses, while the old versions keep using the non `U256` version and make the `derive_base_fee_and_gas_per_pubdata` function in utils dispatch accordingly. Is that reasonable?

Reply:
Yes! IMO it's better to update `L1PeggedBatchFeeModelInput` to use `U256` and introduce struct `L1PeggedBatchFeeModelInputLegacy` with u64 and describe why we need it with a doc-comment. And in old VM version modules you can do `use ...::L1PeggedBatchFeeModelInputLegacy as L1PeggedBatchFeeModelInput` so the diff will be really small

Reply:
After trying out the changes I realized that doing this ends up requiring going all the way to the top to the `Vm` struct and `VmInterface` trait, because it contains an they both contain an `L1BatchEnv`, which holds the gas prices that need to be changed from `u64` to `U256`. 

This is a lot of structs/traits (at least 7) that would now have their `Legacy` version, which is a lot of changes. We can go that route but I want to double check if it's reasonable or if there are other alternatives.

Reply:
@perekopskiy @popzxc can you please check the previous message. Thanks!